### ISSUE TITLE
[feature] Skip unchanged files in conan.tools.files.copy

### ISSUE THREAD

**Issue by Nekto89** [conan-io/conan#16027]
### What is your suggestion?

Hi. I would like to not overwrite files that haven't been changed during call to conan.tools.files.copy. Maybe some optional parameter that would lead to skipping files that have the same timestamp and size? This will lead to less disk operations and will allow to not close applications that are currently using dlls in target folder.

That's how I currently use this method:

```
def generate(self):
        cmake_deps = CMakeDeps(self)
        cmake_deps.configuration = "Release"
        cmake_deps.generate()
        for dep in self.dependencies.values():
            if len(dep.cpp_info.bindirs):
                copy(self, "*.dll", dep.cpp_info.bindirs[0], r"C:/dev/bld/Binaries/Release")
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @Nekto89 

Thanks for the suggestion. We indeed wanted to add some optimization to the ``copy()`` method, not sure if opt-in or opt-out, based on: https://github.com/conan-io/conan/pull/15148, but that was for Conan 1.X, we want this to go to Conan 2.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/16031 was merged, so this will be in Conan 2.3
